ls1intum/Artemis
# [Bugfix] Fix Course Archival Issues

### Checklist

### Motivation and Context


Issues were encountered while archiving courses with many exercises. We saw that multiple zip files from the exported exercises and submissions were deleted before the archive process has finished. This led to an incomplete archive file and many sentry errors since the files are no longer present. 

After some investigations, we think the root cause of the problem is the deletion scheduling which cleans up temporary files included the exported zip after x minutes have passed. 
 
### Description

We made multiple changes:

1. The export functions do not schedule the deletions anymore. This is done by the services that explicitly use the export functionality.
2. We added more exception logging and improved error messages so that we can investigate further if the course archival process still causes problems
3. We attached a small report within the archive file which tells the user how many files were not included

### Steps for Testing

The best way to test this is to have a course that has a couple of exercises of each type. Each exercise should have a couple of submissions. You could technically upload some "large files" e.g a couple of pictures that have ca. 5mb just to stress test the process a little bit.

1. Log in to Artemis
2. Navigate to Course Administration
3. Edit the course and set the date to the past
4. You should see the archive button
5. Click on archive and wait until it finishes.
6. Press the "Download Archive" button that should have appeared and download the file
7. Make sure the submissions are present as well as the template, test, and solution repositories.
8. You should also see a `report.csv` file in the archive which gives a small report
9. You might get some messages at the end of the archive about files that don't exist. This is normal because the archive process does not terminate if files are missing. It simply prints out a warning to the user and dumps those in the `exportErrors.txt` file which is also included within the archive.

### Test Coverage
